### DISCUSSION TITLE
Data in the topics are getting lost

### DISCUSSION THREAD

**Discussion by shreyasarani23** [strimzi/strimzi-kafka-operator#4931] [General]
Hi I have a created a topic on friday i.e [7-05-2021] and was able to consume the messages from topics. And today i.e [10-05-2021] when I try to consume messages it doesn't show me any records. Following is the configuration of my topic.
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaTopic
metadata:
  name: new-topic
  labels:
    strimzi.io/cluster: my-kafka-cluster
spec:
  partitions: 3
  replicas: 3
  config:
    min.insync.replicas: 2
    retention.ms: 7200000
    segment.bytes: 1073741824

following is the kafka broker configuration
    config:
      offsets.topic.replication.factor: 3
      transaction.state.log.replication.factor: 3
      transaction.state.log.min.isr: 2
      log.message.format.version: "2.7"
      auto.leader.rebalance.enable: true
      delete.topic.enable: true

Please tell me if I have to specify any extra configurations.

**Comment by scholzj** [comment]
You configured retention to 2 hours (`retention.ms: 7200000`). There are many things affecting when exactly are the data deleted, but with retention for 2 hours you cannot expect the data from Friday to be present on Monday.

**Reply by shreyasarani23** [reply]
Ohk got it I need to change it some higher value and what are the other things affecting?

**Reply by scholzj** [reply]
If you want the records last longer, you need to increase the retention.

**Reply by shreyasarani23** [reply]
Ok got it along with retention.ms what are the other properties I have to introduce so that data is available for longer period. And with respective to log what are the properties we need to add?

**Reply by scholzj** [reply]
There are many things which define the logic of when messages are deleted - segment size, how often are new segments created etc. I'm sure it is described somewhere in Kafka docs, so please try to find it there.

**Reply by shreyasarani23** [reply]
Ok @scholzj
